lib/utility/src/Logger.cpp

```diff
@@ -1,10 +1,16 @@
 #include "Logger.h"
 
-namespace Utility {
+#include <inttypes.h>
+
+using namespace Utility;
+
 static char buffer[4096];  // Can increase max buffer length if needed
-}
 
-int Utility::Logger::printf(const char* format, ...) {
+#if defined(SWO_LOGGING)
+static int swo_printf(const char* format, ...);  // Only print to SWO (helper for mbed error hook)
+#endif
```

Comment:
cuz it's only used for SWO logging?

Reply:
i dont think we need to be feature flagging these things out since they'll have no chance of conflicting with other things (by the way u named them). lto will remove these automatically during build (once i get around to enabling lto)

Reply:
Ok, currently the unused-function compiler flag raises a compile error if I keep this function but don't use it